src/dungeon.cpp

```diff
@@ -1302,36 +1302,38 @@ static void commandQuit() {
     }
 }
 
-static void commandPreviousMessage() {
-    int maxMessages = MESSAGE_HISTORY_SIZE;
-
+static uint8_t calculateMaxMessageCount () {
+    uint8_t max_messages = MESSAGE_HISTORY_SIZE;
     if (command_count > 0) {
-        maxMessages = command_count;
-        if (maxMessages > MESSAGE_HISTORY_SIZE) {
-            maxMessages = MESSAGE_HISTORY_SIZE;
+        max_messages = command_count;
+        if (max_messages > MESSAGE_HISTORY_SIZE) {
+            max_messages = MESSAGE_HISTORY_SIZE;
         }
```

Comment:
Rather than relying on casting here can we not just handle this in the if statement? What do you think of the following?
```
if (command_count > MESSAGE_HISTORY_SIZE) {
    max_messages = MESSAGE_HISTORY_SIZE;
} else {
    max_messages = (uint8_t) command_count;
}
```